### ISSUE TITLE
v0.9.37 Manual signout causes hard crash after signin

### ISSUE THREAD

**Issue by Firemaw** [playgameservices/play-games-plugin-for-unity#1675]
Using v0.9.37 with v10.2.1 AARs (ads, auth, base, drive, games, nearby, tasks)
support-annotations-24.0.0
support-v4-24.0.0

Occurs on 
Samsung Tab S2 with Android 6.1
Samsung S4 Mini with Android 4.4

Steps:
1. Open game
2. Sign In
3. Sign Out - the errors below occur - this results in a hard crash

```
04-08 22:36:17.860 18775-18801/? I/Unity: Calling Signout in token client
                                           
                                          (Filename: ./artifacts/generated/Android/runtime/DebugBindings.gen.cpp Line: 51)
04-08 22:36:17.900 18775-18801/? I/Unity: Calling Signout in token client
                                           
                                          (Filename: ./artifacts/generated/Android/runtime/DebugBindings.gen.cpp Line: 51)
04-08 22:36:17.950 18775-18953/? I/Unity: Calling Signout in token client
                                           
                                          (Filename: ./artifacts/generated/Android/runtime/DebugBindings.gen.cpp Line: 51)
04-08 22:36:18.421 18775-18801/? W/Unity: *** [Play Games Plugin DLL] 04/08/17 22:36:18 +10:00 ERROR: Error encountered executing 

InternalAuthFinishedCallback. Smothering to avoid passing exception into Native: UnityEngine.AndroidJavaException: 

java.lang.ClassNotFoundException: com.google.games.bridge.TokenFragment
                                          java.lang.ClassNotFoundException: com.google.games.bridge.TokenFragment
                                              at java.lang.Class.classForName(Native Method)
                                              at java.lang.Class.forName(Class.java:251)
                                              at java.lang.Class.forName(Class.java:216)
                                              at dalvik.system.NativeStart.run(Native Method)
                                           Caused by: java.lang.NoClassDefFoundError: com/google/games/bridge/TokenFragment
                                              at java.lang.Class.classForName(Native Method) 
                                              at java.lang.Class.forName(Class.java:251) 
                                              at java.lang.Class.forName(Class.java:216) 
                                              at dalvik.system.NativeStart.run(Native Method) 
                                           Caused by: java.lang.ClassNotFoundException: Didn't find class 

"com.google.games.bridge.TokenFragment" on path: DexPathList[[directory "."],nativeLibraryDirectories=[/vendor/lib, /system/lib]]
                                              at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:67)
                                              at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
                                              at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
                                              at java.lang.Class.classForName(Native Method) 
                                              at java.lang.Class.forName(Class.java:251) 
                                              at java.lang.Class.forName(Class.java:216) 
                                              at dalvik.system.NativeStart.run(Native Method) 

04-08 22:43:22.494 19709-19866/? A/art: art/runtime/java_vm_ext.cc:410]   native: #09 pc 00926644  /data/app/com.x.y-

1/lib/arm/libunity.so (???)
04-08 22:43:22.649 19953-19953/? A/google-breakpad: M 9DB04000 00000000 00FDE000 BBA790D6CC3331DA10B8860AD109367D0 libunity.so
04-08 22:43:22.824 2757-2757/? A/DEBUG:     #14 pc 00926644  /data/app/com.x.y-1/lib/arm/libunity.so

04-08 23:28:51.108 10027-10505/com.x.y A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 10505 (Thread-964)
04-08 23:28:51.113 2702-3592/? D/AudioFlinger: mixer(0xaf740000) throttle end: throttle time(14)
04-08 23:28:51.173 2693-2693/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
04-08 23:28:51.173 2693-2693/? A/DEBUG: Build fingerprint: 'samsung/gts210wifixx/gts210wifi:6.0.1/MMB29K/T810XXU2CPK1:user/release-keys'
04-08 23:28:51.173 2693-2693/? A/DEBUG: Revision: '6'
04-08 23:28:51.173 2693-2693/? A/DEBUG: ABI: 'arm'
04-08 23:28:51.173 2693-2693/? A/DEBUG: pid: 10027, tid: 10505, name: Thread-964  >>> com.x.y <<<
04-08 23:28:51.173 2693-2693/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
```

4. If the game does not crash, Sign In
5. Quit game and close it from task menu
6. Open game - Sign In
7. Sign Out

**Comment by claywilkinson** [comment]
Can you share the full log of reproducing the crash?  Thanks!

**Comment by Firemaw** [comment]
Hi Clay, sure thing, can I send it privately to your email as it may contain sensitive information?

**Comment by mkorotetsky** [comment]
Rerpoduced 100%. Just do logout after login.
[crashlog.txt](https://github.com/playgameservices/play-games-plugin-for-unity/files/907954/crashlog.txt)

**Comment by Kupritas** [comment]
Same here, can reproduce this issue 100% on the Minimal sample. Sign in and sign out makes it crash.

**Comment by claywilkinson** [comment]
I'll update the .unitypackage later today, but here is the fix.  The problem is Signout was using Java from an incorrect thread.  If you replace [AndroidTokenClient.Signout()](https://github.com/playgameservices/play-games-plugin-for-unity/blob/master/source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidTokenClient.cs#L113) with this version it worked for me:
```
public void Signout()
{
    PlayGamesHelperObject.RunOnGameThread(() =>
      {
        Debug.Log("Calling Signout in token client");
        AndroidJavaClass cls = new AndroidJavaClass(TokenFragmentClass);
        cls.CallStatic("signOut");
      });
 }
```

**Comment by Kupritas** [comment]
Thanks Clay!

**Comment by Firemaw** [comment]
Thanks for the fast response Clay! Appreciate it! I can confirm making this change to AndroidTokenClient.cs worked on both my devices.

**Comment by rjeike** [comment]
I wish I would have seen this issue sooner.  Or better yet that the latest library had been updated with this fix.  I spent a whole day troubleshooting this issue, initially under the initial assumption that there was a bug in my application code.  Only when I repo'ed it with the Minimal app did I realize it was bug in the library.  It was so hard to believe such an obvious bug (that Sign Out ALWAYS crashes the app) was present.  Anyway, thanks for the fix!
